cmd/jaeger/internal/integration/kafka_test.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+)
+
+func TestKafkaStorage(t *testing.T) {
+    integration.SkipUnlessEnv(t, "kafka")
+
+    collectorConfig := "../../collector-with-kafka.yaml"
+    ingesterConfig := "../../ingester-remote-storage.yaml"
+
+    collector := &E2EStorageIntegration{
+        ConfigFile: collectorConfig,
+        StorageIntegration: integration.StorageIntegration{
+            GetDependenciesReturnsSource: true,
+            SkipArchiveTest: true,
+        },
+    }
+
+    // Initialize and start the collector
+    collector.e2eInitialize(t, "kafka-collector")
```

Comment:
you may need to introduce a parameter to disable injection of storage cleaner extension, because you may run into port conflict when both binaries will be trying to create a Purge endpoint.

Reply:
yes, was thinking the same. looking into it.

Reply:
added the injectStorageCleaner parameter which when set false disables the injection of the storage cleaner extension.